### ISSUE TITLE
Haven't been able to get this to work on k8s 1.10, weave 2.5 + coredns 

### ISSUE THREAD

**Issue by JaveriaK** [Quentin-M/weave-tc#6]
I'm not quite sure what I'm missing; but adding this to my cluster hasn't helped the dns issue yet.
(kops, aws, 1.10 k8s on jessie, weave-net 2.5, coredns) 

The dns ports used by coredns are default:

```
Pod Template:
  Containers:
   coredns:
    Image:       k8s.gcr.io/coredns:1.1.3
    Ports:       53/UDP, 53/TCP, 9153/TCP
    Host Ports:  0/UDP, 0/TCP, 0/TCP
```

`NET_OVERLAY_IF` is ofcourse also weave.

From within a standard ubuntu container, most of the dns tests generate timeouts within the first few hundred attemps. For example:
```
# - seq 1000 | parallel -j50 --joblog log curl -s https://google.com/ ">" /dev/null; sort -k4 -n log

.....
.....

227	:	1547187611.527	     0.614	0	0	7	0	curl -s https://google.com/ > /dev/null 227
556	:	1547187614.381	     0.615	0	0	7	0	curl -s https://google.com/ > /dev/null 556
25	:	1547187609.772	     0.658	0	0	7	0	curl -s https://google.com/ > /dev/null 25
803	:	1547187616.697	     5.977	0	0	7	0	curl -s https://google.com/ > /dev/null 803
609	:	1547187614.885	     6.029	0	0	7	0	curl -s https://google.com/ > /dev/null 609
603	:	1547187614.846	     6.051	0	0	7	0	curl -s https://google.com/ > /dev/null 603

```

----

The weave-tc container is otherwise up:
```
kubectl log weave-net-2ddv2 -n kube-system -f weave-tc
+ DNSMASQ_PORT=53
+ NET_OVERLAY_IF=weave
+ sysctl -w net.core.default_qdisc=fq_codel
net.core.default_qdisc = fq_codel
+ route
+ grep ^default
+ grep -o [^ ]*$
+ tc qdisc del dev eth0 root
+ true
+ route
+ grep -o [^ ]*$
+ grep ^default
+ tc qdisc add dev eth0 root handle 0: mq
+ ip link
+ grep weave
+ tc qdisc del dev weave root
+ true
+ tc qdisc add dev weave root handle 1: prio bands 2 priomap 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
+ tc qdisc add dev weave parent 1:2 handle 12: fq_codel
+ tc qdisc add dev weave parent 1:1 handle 11: netem delay 4ms 1ms distribution pareto
+ tc filter add dev weave protocol all parent 1: prio 1 handle 0x100/0x100 fw flowid 1:1
+ iptables -A POSTROUTING -t mangle -p udp --dport 53 -m string -m u32 --u32 28 & 0xF8 = 0 --hex-string |00001C0001| --algo bm --from 40 -j MARK --set-mark 0x100/0x100
+ sleep 3600
+ :
+ sleep 3600
```

It is applied like so within the weave-net daemonset:

```
    spec:
      containers:
      - image: qmachu/weave-tc:bd94b89
        imagePullPolicy: IfNotPresent
        name: weave-tc
        resources: {}
        securityContext:
          privileged: true
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /run/xtables.lock
          name: xtables-lock
        - mountPath: /lib/tc
          name: lib-tc
.....
.....
      - hostPath:
          path: /usr/lib/tc
          type: ""
        name: lib-tc

**Comment by Quentin-M** [comment]
Hey!

There are many different types of DNS issues out there (and in the issues linked). As this workaround is specifically for the conntrack race condition, have you confirmed that your DNS issue is actually related to the race condition? To know that, you need to monitor the `insert_failed` counter.

**Comment by JaveriaK** [comment]
So I tried debugging a bit more and did observe the `insert_failed` count increase on test nodes. However, I eventually just ended up implementing a node local dnsmasq solution which fixed all delays completely. 
I see the second kernel fix also just made it to `5.0.0-rc6` a few days ago in case anyone wants to give that a try 
(https://github.com/kubernetes/kubernetes/issues/56903#issuecomment-462252499)

I was likely doing something wrong, but thank you sharing this fix anyhow @Quentin-M .
